pkg/objstore/oci/helper.go

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Apache License 2.0.
+
+package oci
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/oracle/oci-go-sdk/v54/common"
+	"github.com/oracle/oci-go-sdk/v54/objectstorage"
+	"github.com/oracle/oci-go-sdk/v54/objectstorage/transfer"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+func DefaultTransport(config Config) *http.Transport {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+
+		IdleConnTimeout:       time.Duration(config.HTTPConfig.IdleConnTimeout),
+		ResponseHeaderTimeout: time.Duration(config.HTTPConfig.ResponseHeaderTimeout),
+		TLSHandshakeTimeout:   time.Duration(config.HTTPConfig.TLSHandshakeTimeout),
+		ExpectContinueTimeout: time.Duration(config.HTTPConfig.ExpectContinueTimeout),
+		TLSClientConfig:       &tls.Config{InsecureSkipVerify: config.HTTPConfig.InsecureSkipVerify},
+		MaxIdleConns:          config.HTTPConfig.MaxIdleConns,
+		MaxIdleConnsPerHost:   config.HTTPConfig.MaxIdleConnsPerHost,
+		MaxConnsPerHost:       config.HTTPConfig.MaxConnsPerHost,
+		DisableCompression:    config.HTTPConfig.DisableCompression,
+	}
+}
+
+func getNamespace(client objectstorage.ObjectStorageClient, requestMetadata common.RequestMetadata) (namespace *string, err error) {
+	response, err := client.GetNamespace(
+		context.Background(),
+		objectstorage.GetNamespaceRequest{RequestMetadata: requestMetadata},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return response.Value, nil
+}
+
+func getObject(ctx context.Context, bkt Bucket, objectName string, byteRange string) (response objectstorage.GetObjectResponse, err error) {
+	// path should not be empty
```

Comment:
Nit: let's remove this, we can see what the `if` is doing

Reply:
Removed.